lang/cs/Org.Apache.REEF.Tang/Implementations/ClassHierarchy/ClassHierarchyImpl.cs

```diff
@@ -167,6 +167,18 @@ public INode RegisterType(Type type)
                         RegisterType(constructorArg.Gettype());  // Gettype returns param's Type.fullname
                         if (constructorArg.GetNamedParameterName() != null)
                         {
+                            var pn = RegisterType(constructorArg.GetNamedParameterName());
+                            
+                            if (!(pn is INamedParameterNode))
+                            {
+                                string fullNamedParameterName = constructorArg.GetNamedParameterName();
```

Comment:
You can replace all with this line:
 string message = string.Format(CultureInfo.CurrentCulture, "The class {0}, defined in the constructor of {1}, should not be defined as a NamedParameter.", constructorArg.GetNamedParameterName(), constructorDef.GetClassName());

And the error message will be ends up like:
The class Org.Apache.REEF.Tang.Tests.ClassHierarchy.ClazzA+ClazzB, Org.Apache.REEF.Tang.Tests, Version=0.15.0.0, Culture=neutral, PublicKeyToken=c27bf5b2e9a7ddb9, defined in the constructor of Org.Apache.REEF.Tang.Tests.ClassHierarchy.ClazzA, Org.Apache.REEF.Tang.Tests, Version=0.15.0.0, Culture=neutral, PublicKeyToken=c27bf5b2e9a7ddb9, should not be defined as a NamedParameter.